packages/cli/lib/charm.ts

```diff
@@ -0,0 +1,207 @@
+import { ANYONE, Identity, Session } from "@commontools/identity";
+import { ensureDir } from "@std/fs";
+import { loadIdentity } from "./identity.ts";
+import {
+  Cell,
+  getEntityId,
+  Recipe,
+  RecipeMeta,
+  Runtime,
+} from "@commontools/runner";
+import { StorageManager } from "@commontools/runner/storage/cache";
+import {
+  Charm,
+  CharmManager,
+  compileRecipe,
+  getRecipeIdFromCharm,
+} from "@commontools/charm";
+import { join } from "@std/path";
+
+export interface SpaceConfig {
+  apiUrl: string;
+  space: string;
+  identity: string;
+}
+
+export interface CharmConfig extends SpaceConfig {
+  charm: string;
+}
+
+function parseSpace(
+  space: string,
+): string {
+  if (space.startsWith("did:key:")) {
+    // Need to be able to resolve a did key to a name, based
+    // on current Session requirements.
+    throw new Error("`space` as a DID key is not yet supported.");
+  }
+  if (space.startsWith("~")) {
+    // Need to be able to resolve a private space to a did key, based
+    // on current Session requirements.
+    throw new Error("`space` must not be a private space.");
+  }
+  return space;
+}
+
+function getCharmId(charm: Cell<Charm>): string {
+  const id = getEntityId(charm)?.["/"];
+  if (!id) {
+    throw new Error("Could not get an ID from a Cell<Charm>");
+  }
+  return id;
+}
+
+async function makeSession(config: SpaceConfig): Promise<Session> {
+  if (config.space.startsWith("did:key")) {
+    throw new Error("DID key spaces not yet supported.");
+  }
+  const root = await loadIdentity(config.identity);
+  const account = config.space.startsWith("~")
+    ? root
+    : await Identity.fromPassphrase(ANYONE);
+  const user = await account.derive(config.space);
```

Comment:
it seems odd that the user is derived from the space ...  but it is what we do in jumble https://github.com/commontoolsinc/labs/blob/main/packages/jumble/src/contexts/AuthenticationContext.tsx#L113